sandbox.go

```diff
@@ -676,23 +676,68 @@ func (sb *sandbox) SetKey(basePath string) error {
 
 func (sb *sandbox) EnableService() error {
 	logrus.Debugf("EnableService %s START", sb.containerID)
+
 	for _, ep := range sb.getConnectedEndpoints() {
-		if ep.enableService(true) {
+		if !ep.isServiceEnabled() {
 			if err := ep.addServiceInfoToCluster(sb); err != nil {
-				ep.enableService(false)
 				return fmt.Errorf("could not update state for endpoint %s into cluster: %v", ep.Name(), err)
 			}
+			// enable service on the endpoint copy in the sandbox
+			ep.enableService(true)
```

Comment:
looks like I deleted a line. One was to update the store and one to update in mem copy. Let me rework this and test it out.